Terminal.Gui/ConsoleDrivers/WindowsDriver.cs

```diff
@@ -970,70 +990,88 @@ KeyCode MapKey (WindowsConsole.ConsoleKeyInfoEx keyInfoEx)
+						// any info that the previous event was a deadkey.
+						// TODO: We *could* set a global with charVal and then, on the next KeyDown combine them,
+						// but that's super-complex, and WT does not support this either. 
+						// This means Ctrl-DeadKey input is not supported in Terminal.Gui apps
+						return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)charVal);
+					}
+					return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)charVal);
+				} else {
+					return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)keyInfo.Key);
+				}
+			} else {
+				// Strip off Shift - We got here because they KeyChar from Windows is the shifted char (e.g. "�") and passing on Shift
+				// would be redundant.
+				return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers & ~ConsoleModifiers.Shift, (KeyCode)keyInfo.KeyChar );
+			}
+			break;
 		}
 
 		var key = keyInfo.Key;
 
-		if (key >= ConsoleKey.A && key <= ConsoleKey.Z) {
-			var delta = key - ConsoleKey.A;
+		// A..Z are special cased:
+		// - Alone, they represent lowercase a...z
+		// - With ShiftMask they are A..Z
+		// - If CapsLock is on the above is reversed.
+		// - If Alt and/or Ctrl are present, treat as upper case
+		if (key is >= ConsoleKey.A and <= ConsoleKey.Z) {
 			if (keyInfo.Modifiers == ConsoleModifiers.Control) {
-				return (KeyCode)(((uint)KeyCode.CtrlMask) | ((uint)KeyCode.A + delta));
+				return (KeyCode)(((uint)KeyCode.CtrlMask) | (uint)key);
 			}
 			if (keyInfo.Modifiers == ConsoleModifiers.Alt) {
-				return (KeyCode)(((uint)KeyCode.AltMask) | ((uint)KeyCode.A + delta));
+				return (KeyCode)(((uint)KeyCode.AltMask) | ((uint)key));
 			}
 			if (keyInfo.Modifiers == (ConsoleModifiers.Shift | ConsoleModifiers.Alt)) {
-				return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)((uint)KeyCode.A + delta));
-			}
-			if (keyInfo.Modifiers == (ConsoleModifiers.Control | ConsoleModifiers.Shift)) {
-				return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)((uint)KeyCode.A + delta));
-			}
-			if (!keyInfo.Modifiers.HasFlag (ConsoleModifiers.Shift)
-				&& (keyInfo.Modifiers & (ConsoleModifiers.Alt | ConsoleModifiers.Control)) != 0) {
-				if (keyInfo.KeyChar == 0 || (keyInfo.KeyChar != 0 && keyInfo.KeyChar >= 1 && keyInfo.KeyChar <= 26)) {
-					return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)((uint)KeyCode.A + delta));
-				}
+				return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)((uint)key));
 			}
 
-			if (((keyInfo.Modifiers == ConsoleModifiers.Shift) ^ (keyInfoEx.CapsLock))) {
-				if (keyInfo.KeyChar >= (uint)KeyCode.A && keyInfo.KeyChar <= (uint)KeyCode.Z) {
-					return (KeyCode)((uint)KeyCode.A + delta) | KeyCode.ShiftMask;
-				} else {
-					return (KeyCode)keyInfo.KeyChar | KeyCode.ShiftMask;
-				}
+			if ((keyInfo.Modifiers & (ConsoleModifiers.Alt | ConsoleModifiers.Control)) != 0) {
```

Comment:
This statement isn't doing what is supposed to do that is only execute if modifiers are `Ctrl+Alt`.
This isn't correct because it will return true if we press Ctrl+Shift+E.

![image](https://github.com/BDisp/Terminal.Gui/assets/13117724/8fce1f14-9934-4caa-ae04-1212f341822b)

Reply:
Sorry I was wrong here. It have to execute anything that has the Ctrl key.